analyzers/src/SonarAnalyzer.CSharp.Styling/Rules/LambdaParameterName.cs

```diff
@@ -0,0 +1,38 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp.Styling;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class LambdaParameterName : StylingAnalyzer
+{
+    public LambdaParameterName() : base("T0010", "Use 'x' for the lambda parameter name.") { }
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterNodeAction(c =>
+            {
+                var parameter = ((SimpleLambdaExpressionSyntax)c.Node).Parameter;
+                if (parameter.Identifier.ValueText != "x" && c.Node.Parent.FirstAncestorOrSelf<LambdaExpressionSyntax>() is null)
```

Comment:
This lambda would raise, and we have quite a lot of these in the code base for context action.

Either we do not raise on lambda with an explicit body or add an exception for SonarReportingContext parameter type.

Reply:
That's a good catch